pytest/lib/mocknet.py

```diff
@@ -327,6 +336,103 @@ def stop_node(node):
+        genesis_config['chain_id'] = "mocknet"
+        genesis_config['total_supply'] = str(
+            int(MASTER_BALANCE) + int(TREASURY_BALANCE) + len(nodes) * (int(VALIDATOR_BALANCE) + int(STAKED_BALANCE)))
+        genesis_config['records'] = []
+        genesis_config['validators'] = []
+        genesis_config['records'].append({"Account": {"account_id": TREASURY_ACCOUNT,
+                                                      "account": {"amount": TREASURY_BALANCE, "locked": "0",
+                                                                  "code_hash": "11111111111111111111111111111111",
+                                                                  "storage_usage": 0, "version": "V1"}}})
+        genesis_config['records'].append({"AccessKey": {"account_id": TREASURY_ACCOUNT, "public_key": PUBLIC_KEY,
+                                                        "access_key": {"nonce": 0, "permission": "FullAccess"}}})
+        genesis_config['records'].append({"Account": {"account_id": MASTER_ACCOUNT,
+                                                      "account": {"amount": MASTER_BALANCE, "locked": "0",
+                                                                  "code_hash": "11111111111111111111111111111111",
+                                                                  "storage_usage": 0, "version": "V1"}}})
+        genesis_config['records'].append({"AccessKey": {"account_id": MASTER_ACCOUNT, "public_key": PUBLIC_KEY,
+                                                        "access_key": {"nonce": 0, "permission": "FullAccess"}}})
+        for node in nodes:
+            genesis_config['records'].append({"Account": {"account_id": node_account_name(node.instance_name),
+                                                          "account": {"amount": VALIDATOR_BALANCE,
+                                                                      "locked": STAKED_BALANCE,
+                                                                      "code_hash": "11111111111111111111111111111111",
+                                                                      "storage_usage": 0, "version": "V1"}}})
+            genesis_config['records'].append({"AccessKey": {"account_id": node_account_name(node.instance_name),
+                                                            "public_key": PUBLIC_KEY,
+                                                            "access_key": {"nonce": 0, "permission": "FullAccess"}}})
+            genesis_config['validators'].append(
+                {"account_id": node_account_name(node.instance_name), "public_key": PUBLIC_KEY,
+                 "amount": STAKED_BALANCE})
+    with open(mocknet_genesis_filename, 'w') as f:
+        f.write(json.dumps(genesis_config, indent=2))
+
+
+def update_boot_nodes(nodes):
+    assert (len(nodes) > 0)
+    first_node = nodes[0]
+    mocknet_config_filename = '/tmp/mocknet_config.json'
+    first_node_key_filename = '/tmp/mocknet_node_key.json'
```

Comment:
Again, prefer `NamedTemporaryFile` since it respects `TEMPDIR` environment variable and cleans up the files.